client/src/amplify-ui/block-switcher/block-switcher.tsx

```diff
@@ -81,42 +110,38 @@ export class AmplifyBlockSwitcher {
     }
   }
 
-  createActiveChildISetter = (i: number) => (_event: Event): void => {
-    this.activeChildI = i;
-    const current = this.tabHeadings[i];
-    this.setNewSelectedTabHeadings(current);
-  };
+  handleButtonClick(newActiveChild: number) {
```

Comment:
This reads more cleanly than the curried version we had before imo 😄

Reply:
Agree. And wanted to eliminate as much potential for stale closures as possible since I have a child's understanding of how webcomponents work